keras/layers/rnn/base_rnn.py

```diff
@@ -905,8 +905,23 @@ def reset_states(self, states=None):
                         self.variable_dtype or backend.floatx(),
                     )
                 )
+
+            def gen_state_names():
```

Comment:
Since we are adding _gen_state_names() to the LSTM, how about put this as a private method on the base_rnn as _gen_state_names() base implementation?